torch/_tensor_str.py

```diff
@@ -65,132 +65,130 @@ def set_printoptions(
         PRINT_OPTS.linewidth = linewidth
 
 
-def _get_min_log_scale():
-    min_positive = float_info.min * float_info.epsilon  # get smallest denormal
-    if min_positive == 0:  # use smallest normal if DAZ/FTZ is set
-        min_positive = float_info.min
-    return math.ceil(math.log(min_positive, 10))
+class _Formatter(object):
+    def __init__(self, tensor):
+        self.floating_dtype = tensor.dtype.is_floating_point
+        self.int_mode = True
+        self.sci_mode = False
+        self.max_width = 1
+
+        if not self.floating_dtype:
+            copy = torch.LongTensor(tensor.size()).copy_(tensor).view(tensor.nelement())
+            for value in copy.tolist():
+                value_str = '{}'.format(value)
+                self.max_width = max(self.max_width, len(value_str))
 
+        else:
+            copy = torch.DoubleTensor(tensor.size()).copy_(tensor).view(tensor.nelement())
```

Comment:
nit: you could move these two lines out of the if/else.

Reply:
In the if statement I'm actually making a LongTensor instead of a Double tensor, so the two statements are different.

Reply:
ah, right, my mistake.  It still might be clearer if you just passed a dtype (based on int_mode) to torch.empty, but this is nit picky.